### PR TITLE
Add DebugInfoEIS option to allow setting desired extended instruction set for debug info

### PR DESCRIPTION
**Pull request by NikitaRudenkoIntel**
This option required for compatibility with older version of translator

### CODE REVIEW

**Review by AlexeySachkov — APPROVED** [review]
LGTM, but I would like to hear feedback from others as well

File: tools/llvm-spirv/llvm-spirv.cpp [hunk 1]
```diff
@@ -183,6 +183,16 @@ cl::opt<bool> SPIRVAllowUnknownIntrinsics(
     cl::desc("Unknown LLVM intrinsics will be translated as external function "
              "calls in SPIR-V"));
 
+static cl::opt<SPIRV::DebugInfoEIS> DebugEIS(
+    "debug-info-eis", cl::desc("Set debug info EIS:"),
```
Comment (AlexeySotkin) [inline tools/llvm-spirv/llvm-spirv.cpp:187]: Typically SPIR-V translator options begin with `-spirv`.
"EIS" is not quite obvious abbreviation. Can we use "version" instead?
```suggestion
    "spirv-debug-info-version", cl::desc("Set SPIR-V debug info version:"),
```

File: tools/llvm-spirv/llvm-spirv.cpp [hunk 2]
```diff
@@ -183,6 +183,16 @@ cl::opt<bool> SPIRVAllowUnknownIntrinsics(
...
+    cl::init(SPIRV::DebugInfoEIS::OpenCL_DebugInfo_100),
+    cl::values(clEnumValN(SPIRV::DebugInfoEIS::SPIRV_Debug, "legacy",
+                          "SPIRV.debug EIS as legacy option for compatibility "
+                          "with older versions of translator"),
```
Comment (AlexeySotkin) [inline tools/llvm-spirv/llvm-spirv.cpp:191]:
```suggestion
                          "Emit debug info compliant with the SPIRV.debug extended instruction set. "
                          "This option is used for compatibility with older versions of the translator"),
```

File: tools/llvm-spirv/llvm-spirv.cpp [hunk 3]
```diff
@@ -183,6 +183,16 @@ cl::opt<bool> SPIRVAllowUnknownIntrinsics(
...
+               clEnumValN(SPIRV::DebugInfoEIS::OpenCL_DebugInfo_100, "ocl-100",
+                          "OpenCL.DebugInfo.100 EIS. It is required for "
+                          "SPIRV-Tools compatibility")));
```
Comment (AlexeySotkin) [inline tools/llvm-spirv/llvm-spirv.cpp:194]:
```suggestion
                          "Emit debug info compliant with the OpenCL.DebugInfo.100 extended instruction set. "
                          "This version of SPIR-V debug info format is compatible with the SPIRV-Tools")));
```

File: tools/llvm-spirv/llvm-spirv.cpp [hunk 4]
```diff
@@ -568,6 +578,15 @@ int main(int Ac, char **Av) {
     }
   }
 
+  if (DebugEIS.getNumOccurrences() != 0) {
+    if (IsReverse) {
+      errs() << "Note: --debug-info-eis option ignored as it only "
```
Comment (AlexeySotkin) [inline tools/llvm-spirv/llvm-spirv.cpp:583]:
```suggestion
      errs() << "Note: --spirv-debug-info-version option ignored as it only "
```

File: tools/llvm-spirv/llvm-spirv.cpp (hunk 1 above)
Comment (NikitaRudenkoIntel) [reply tools/llvm-spirv/llvm-spirv.cpp]: Fixed

File: tools/llvm-spirv/llvm-spirv.cpp (hunk 2 above)
Comment (NikitaRudenkoIntel) [reply tools/llvm-spirv/llvm-spirv.cpp]: Fixed

File: tools/llvm-spirv/llvm-spirv.cpp (hunk 3 above)
Comment (NikitaRudenkoIntel) [reply tools/llvm-spirv/llvm-spirv.cpp]: Fixed

File: tools/llvm-spirv/llvm-spirv.cpp (hunk 4 above)
Comment (NikitaRudenkoIntel) [reply tools/llvm-spirv/llvm-spirv.cpp]: Fixed

**Review by AlexeySotkin — APPROVED** [review]

**Pull request merged**